### PR TITLE
feat: use serveframework with self signed certificates

### PR DESCRIPTION
**Pull request by marianfoo**

### CODE REVIEW

File: packages/ui5-middleware-serveframework/lib/index.js [hunk 1]
```diff
@@ -73,16 +74,24 @@ module.exports = async ({ log, options, middlewareUtil }) => {
...
-			const proxyUrl = getProxyForUrl(baseUrl);
 			const { HttpsProxyAgent } = await import("https-proxy-agent");
-			const agent = proxyUrl ? new HttpsProxyAgent(proxyUrl) : undefined;
-			effectiveOptions.debug && log.info(`[${baseUrl}] Proxy: ${proxyUrl ? proxyUrl : "n/a"}`);
+			const https = await import("https");
+
+			const proxyUrl = getProxyForUrl(baseUrl);
+			const agentOptions = { rejectUnauthorized: effectiveOptions.strictSSL };
+
+			const agent = proxyUrl ? new HttpsProxyAgent({ ...new URL(proxyUrl), ...agentOptions }) : new https.Agent(agentOptions);
```
Comment (petermuessig) [inline packages/ui5-middleware-serveframework/lib/index.js:83]: `...new URL(proxyUrl)` - why object destructuring?

File: packages/ui5-middleware-serveframework/lib/index.js [hunk 2]
```diff
@@ -73,16 +74,24 @@ module.exports = async ({ log, options, middlewareUtil }) => {
...
-			const proxyUrl = getProxyForUrl(baseUrl);
 			const { HttpsProxyAgent } = await import("https-proxy-agent");
-			const agent = proxyUrl ? new HttpsProxyAgent(proxyUrl) : undefined;
-			effectiveOptions.debug && log.info(`[${baseUrl}] Proxy: ${proxyUrl ? proxyUrl : "n/a"}`);
+			const https = await import("https");
+
+			const proxyUrl = getProxyForUrl(baseUrl);
+			const agentOptions = { rejectUnauthorized: effectiveOptions.strictSSL };
+
+			const agent = proxyUrl ? new HttpsProxyAgent(Object.assign(new URL(proxyUrl), agentOptions)) : new https.Agent(agentOptions);
```
Comment (petermuessig) [inline packages/ui5-middleware-serveframework/lib/index.js:83]: `Object.assign`? 

image

Comment (petermuessig) [reply packages/ui5-middleware-serveframework/lib/index.js:83]: Shouldn't it be: `new HttpsProxyAgent(new URL(proxyUrl), agentOptions)`?

Comment (marianfoo) [reply packages/ui5-middleware-serveframework/lib/index.js]: true, maybe i was doing something when i was trying some stuff out

File: packages/ui5-middleware-serveframework/lib/index.js [hunk 3]
```diff
@@ -73,16 +74,24 @@ module.exports = async ({ log, options, middlewareUtil }) => {
 
 			// support for coporate proxies
 			const { getProxyForUrl } = await import("proxy-from-env");
-			const proxyUrl = getProxyForUrl(baseUrl);
 			const { HttpsProxyAgent } = await import("https-proxy-agent");
-			const agent = proxyUrl ? new HttpsProxyAgent(proxyUrl) : undefined;
-			effectiveOptions.debug && log.info(`[${baseUrl}] Proxy: ${proxyUrl ? proxyUrl : "n/a"}`);
+			const https = await import("https");
```
Comment (petermuessig) [inline packages/ui5-middleware-serveframework/lib/index.js:78]: Native modules should be imported on top of the file with require and you can use object destructuring to directly access the agent:

```js
const { Agent: HttpsAgent } = require("https");
```

File: packages/ui5-middleware-serveframework/lib/index.js [hunk 4]
```diff
@@ -73,16 +74,24 @@ module.exports = async ({ log, options, middlewareUtil }) => {
...
+
+			const proxyUrl = getProxyForUrl(baseUrl);
```
Comment (petermuessig) [inline packages/ui5-middleware-serveframework/lib/index.js:80]: please add a small comment before this line:

```js
// detect and configure proxy agent
```

File: packages/ui5-middleware-serveframework/lib/index.js [hunk 5]
```diff
@@ -73,16 +74,24 @@ module.exports = async ({ log, options, middlewareUtil }) => {
...
+			const agentOptions = { rejectUnauthorized: effectiveOptions.strictSSL };
+
+			const agent = proxyUrl ? new HttpsProxyAgent(new URL(proxyUrl), agentOptions) : new https.Agent(agentOptions);
```
Comment (petermuessig) [inline packages/ui5-middleware-serveframework/lib/index.js:83]: You don't need to create a URL object - just pass the proxy URL:

```js
const agent = proxyUrl ? new HttpsProxyAgent(proxyUrl, agentOptions) : new HttpsAgent(agentOptions);
```

File: packages/ui5-middleware-serveframework/lib/index.js [hunk 6]
```diff
@@ -73,16 +74,24 @@ module.exports = async ({ log, options, middlewareUtil }) => {
...
+				log.info(`[${baseUrl}] Proxy: ${proxyUrl || "n/a"}, strictSSL: ${effectiveOptions.strictSSL}`);
+			}
 
 			// fetch the version information for the concrete version from CDN
 			const fetch = (await import("node-fetch")).default;
-			await fetch(`${baseUrl}/${frameworkVersion}/resources/sap-ui-version.json`, { agent })
-				.then((res) => res.json())
-				.then((json) => writeFile(versionInfoFile, JSON.stringify(json), { encoding: "utf-8" }));
+			const versionUrl = `${baseUrl}/${frameworkVersion}/resources/sap-ui-version.json`;
+			const response = await fetch(versionUrl, { agent });
```
Comment (petermuessig) [inline packages/ui5-middleware-serveframework/lib/index.js:92]: Now I understand it - here you pass the native HttpsAgent with the option `rejectUnauthorized` and this disables the ssl verification for fetch

👍

**Review by petermuessig — CHANGES_REQUESTED** [review]

**Review by petermuessig — APPROVED** [review]
LGTM now, thx for your contribution!

**Comment by marianfoo** [comment]
Thanks for merging!

**Pull request merged**